db.go

```diff
@@ -69,6 +69,14 @@ func (db *DB) Insert(collectionName string, docs ...*Document) error {
 	return db.engine.Insert(collectionName, insertDocs...)
 }
 
+// Save or update a document
+func (db *DB) Save(collectionName string, doc *Document) error {
+	if doc.Get(objectIdField) == "" {
```

Comment:
use doc.Has(objectIdField) instead of checking if the result value of **Get** is equal to the empty string.